sycl/test/basic_tests/buffer/subbuffer.cpp

```diff
@@ -0,0 +1,89 @@
+// RUN: %clang -std=c++11 -fsycl %s -o %t.out -lstdc++ -lOpenCL
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+//==---------- subbuffer.cpp --- sub-buffer basic test ---------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+int main() {
+
+  bool Failed = false;
+  // Basic test case
+  {
+    const int M = 6;
+    const int N = 7;
+    int Result[M][N] = {0};
+    {
+      auto OrigRange = range<2>(M, N);
+      buffer<int, 2> Buffer(OrigRange);
+      Buffer.set_final_data((int *)Result);
+      auto Offset = id<2>(1, 1);
+      auto SubRange = range<2>(M - 2, N - 2);
+      queue MyQueue;
+      buffer<int, 2> SubBuffer(Buffer, Offset, SubRange);
+      MyQueue.submit([&](handler &cgh) {
+        auto B = SubBuffer.get_access<access::mode::read_write>(cgh);
+        cgh.parallel_for<class Subbuf_test>(SubRange,
+                                            [=](id<2> Index) { B[Index] = 1; });
+      });
+    }
+
+    // Check that we filled correct subset of buffer:
+    // 0000000     0000000
+    // 0000000     0111110
+    // 0000000 --> 0111110
+    // 0000000     0111110
+    // 0000000     0111110
+    // 0000000     0000000
+
+    for (size_t i = 0; i < M; ++i) {
+      for (size_t j = 0; j < N; ++j) {
+        size_t Expected =
+            ((i == 0) || (i == M - 1) || (j == 0) || (j == N - 1)) ? 0 : 1;
```

Comment:
Not clear why a `size_t`. Actually a `bool` looks fine to me.
```
auto Expected = !(i == 0) || (i == M - 1) || (j == 0) || (j == N - 1));
```
Replacing `auto` by `int` is also possible if you think it would cause the reader more time to understand the next line...